includes/command.inc

```diff
@@ -696,9 +696,9 @@ function drush_redispatch_get_options() {
   unset($cli_context['php']);
   unset($cli_context['php-options']);
   // cli overrides and command specific
-  $options = $cli_context + drush_get_context('specific');
+  $options_soup = $cli_context + drush_get_context('specific') + drush_get_context('alias');
   $global_option_list = drush_get_global_options(FALSE);
-  foreach (drush_get_context('alias') as $key => $value) {
+  foreach ($options_soup as $key => $value) {
```

Comment:
This introduces a minor change in behavior in addition to the bugfix.

Bugfix: corrects priority of options, so cli options override alias options, as they should. This part of the change is good.
Behavior change: Non-global cli options that are NOT defined as part of the command were formerly passed along to the remote side, but now are silently dropped.  This is probably a bad thing.

I'll try pushing another commit here to fix this.